subiquitycore/async_helpers.py

```diff
@@ -45,6 +48,14 @@ def schedule_task(coro, propagate_errors=True):
 background_tasks = set()
 
 
+def connect_async_signal(obj, name, callback, **kwargs):
```

Comment:
I don't think we can do much better than `Callable[..., Awaitable[Any]` since the handler can take a variadic number of arguments. At least, it "forces" the use of a coroutine function rather than a function.